.github/workflows/scripts/strain_validation/validate_schema.py

```diff
@@ -0,0 +1,150 @@
+        ),
+        Optional(Regex(r"^PICASSO_.+_DPKG$")): {
+            "name": And(str, len),
+            "repo": And(str, len),
+            "version": And(str, len)
+        },
+        Optional("PICASSO_THEMES"): And(
+            list,
+            len,
+            [
+                {
+                    "name": And(str, len),
+                    "repo": And(str, len),
+                    "version": And(str, len)
+                }
+            ]
+        ),
+        Optional("PICASSO_THEMES_NAME"): And(
+            list,
+            len,
+            lambda x: all(isinstance(item, str) and item for item in x),
+            error=EMPTY_LIST_ERROR_MSG.format("PICASSO_THEMES_NAME"),
+        ),
+        Optional("PICASSO_THEME_DIRS"): And(
+            list,
+            len,
+            lambda x: all(isinstance(item, str) and item for item in x),
+            error=EMPTY_LIST_ERROR_MSG.format("PICASSO_THEME_DIRS"),
+        )
+    },
+    ignore_extra_keys=True
+)
+
+def validate_with_warnings(data: Dict[str, Any]) -> bool:
+    """
+    Validate the data against the strain schema and log warnings for missing optional keys.
+
+    Args:
+        data (Dict[str, Any]): The loaded YAML data to validate.
+
+    Returns:
+        bool: True if validation is successful; otherwise, False.
+
+    Logs warnings for missing optional keys such as PICASSO_THEMES and PICASSO_THEMES_NAME.
+    """
+    try:
+        STRAIN_SCHEMA.validate(data)
+        if not data.get("PICASSO_THEMES"):
+            LOG.warning("No PICASSO_THEMES key found; themes will not be enabled.")
+        if not data.get("PICASSO_THEMES") and not data.get("PICASSO_THEMES_NAME"):
+            LOG.warning("No PICASSO_THEMES_NAME key found; default themes will be used.")
+        LOG.info("Strain syntax and structure validation completed successfully.")
+        return True
+    except SchemaError as e:
+        LOG.error("Schema validation failed: %s", e)
+        return False
+
+def validate_yaml_file(yaml_file: TextIOWrapper) -> bool:
+    """
+    Load and validate YAML file structure against the defined schema.
+
+    Args:
+        yaml_file (TextIOWrapper): Opened YAML file for reading.
+
+    Returns:
+        bool: True if YAML content is valid; otherwise, False.
+
+    Logs syntax errors in the YAML structure.
+    """
+    try:
+        config_yml = yaml.safe_load(yaml_file)
+        return validate_with_warnings(config_yml)
+    except yaml.YAMLError as yaml_error:
+        LOG.error("YAML syntax error: %s", yaml_error)
+    return False
+
+def main(argv: Sequence[str] | None = None) -> int:
+    """
+    Execute syntax checks on a configuration file for strains.
+
+    Args:
+        argv (Sequence[str] | None): Command-line arguments.
+
+    Returns:
+        int: 0 if configuration file is valid; 1 if invalid.
+    """
+    parser = argparse.ArgumentParser(description="Validate YAML file syntax and strain schema.")
```

Comment:
Could we validate the arguments in another function similar to how it is done [here](https://github.com/eduNEXT/picasso/blob/d857639fcf73d309721b5479403a280747a82891/.github/workflows/scripts/get_tutor_config.py#L72-L96)?